src/bentoml/_internal/io_descriptors/json.py

```diff
@@ -1,15 +1,16 @@
 from __future__ import annotations
 
+import sys
 import json
 import typing as t
 import logging
 import dataclasses
 
 import attr
-from typing_extensions import TypedDict
-from typing_extensions import is_typeddict
+from pydantic.typing import is_typeddict_special
 from starlette.requests import Request
 from starlette.responses import Response
+from pydantic.annotated_types import is_legacy_typeddict
```

Comment:
We don't want to eager import like this? pydantic is still optional for `bentoml.io.JSON`.